crates/hir/src/lib.rs

```diff
@@ -685,6 +685,26 @@ impl Module {
                     },
                 ));
 
+                let reduntant: Vec<_> = impl_assoc_items_scratch
+                    .iter()
+                    .filter(|(id, name)| {
+                        !items.iter().any(|(impl_name, impl_item)| {
+                            discriminant(impl_item) == discriminant(id) && impl_name == name
+                        })
+                    })
+                    .map(|(item, name)| (name.clone(), AssocItem::from(*item)))
+                    .collect();
```

Comment:
```suggestion
                    .map(|(item, name)| (name.clone(), AssocItem::from(*item)));
```
No need to collect here